src/wallet/coinselection.cpp

```diff
@@ -194,7 +194,7 @@ util::Result<SelectionResult> SelectCoinsBnB(std::vector<OutputGroup>& utxo_pool
     for (const size_t& i : best_selection) {
         result.AddInput(utxo_pool.at(i));
     }
-    result.ComputeAndSetWaste(cost_of_change, cost_of_change, CAmount{0});
+    result.RecalculateWaste(cost_of_change, cost_of_change, CAmount{0});
```

Comment:
Could set the second parameter, `change_cost` to 0 instead of providing `cost_of_change`. The parameter is only used if there is change and BnB goal is to not generate one.

Reply:
I guess, but the point of this PR is that `change_cost` no longer needs to be set to 0 to indicate that no change is being created.

Reply:
Unfortunately, we don’t have the `change_fee` or `feerate` readily available in the BnB function. The following passes the tests, though:

```cpp
-    result.RecalculateWaste(cost_of_change, cost_of_change, CAmount{0});
+    /* Calculate the fee for a P2WPKH change output */
+    if (utxo_pool.at(0).long_term_fee) {
+        const CFeeRate feerate = utxo_pool.at(0).m_long_term_feerate * (utxo_pool.at(0).fee / utxo_pool.at(0).long_term_fee);
+        const CAmount change_fee = feerate.GetFee(31);
+        result.RecalculateWaste(cost_of_change, cost_of_change, change_fee);
+    } else {
+        /* Use 1 sat/vB and 31 vB in case no long_term_feerate is set */
+        result.RecalculateWaste(cost_of_change, cost_of_change, /*change_fee=*/CAmount{31});
+    }
```

Reply:
Let me know if you have a suggestion how I should amend the code.